apis/externalsecrets/v1beta1/externalsecret_types.go

```diff
@@ -303,6 +303,11 @@ type ExternalSecretStatus struct {
 
 	// +optional
 	Conditions []ExternalSecretStatusCondition `json:"conditions,omitempty"`
+
+	// +nullable
+	// +optional
+	// leaseTimeout is the time and date the data fetched by the provider will expire
+	LeaseTimeout metav1.Time `json:"leaseTimeout,omitempty"`
```

Comment:
I'm not too sure if i want to add this field to the `ExternalSecret`. It seems too specific for me: this only makes sense if a secret expires like short-lived credentials. But the `ExternalSecret` - in my mind - is conceptually closer to "fetch key-value pairs from a remote API" than that. And for that use-case we don't care about expiry times because the secret lifecycle is managed outside.

I make the following assumptions: 
1. lease times are usually static
2. the `ExternalSecret` user knows the lease time (or is able to get that information)
3. the lease time begins when we fetch a secret

With that a user can leverage the `spec.refreshInterval` to rotate the credentials before they expire. E.g. `lease=12h`, `refreshInterval=10h`. I think in this scenario we wouldn't even need a `spec.leaseTimeout`.

We're currently discussing a different approach to allow specific configuration in the `ExternalSecret` that @gusfcarvalho mentioned further up. This should give us the flexibility to provide custom configuration (like a leaseTimeout) with an ExternalSecret: https://github.com/external-secrets/external-secrets/pull/1338